zcash_primitives/src/zip32.rs

```diff
@@ -409,6 +446,38 @@ impl ExtendedSpendingKey {
     pub fn default_address(&self) -> (DiversifierIndex, PaymentAddress) {
         ExtendedFullViewingKey::from(self).default_address()
     }
+
+    pub fn derive_internal(&self) -> Self {
+        let i = {
+            let fvk = FullViewingKey::from_expanded_spending_key(&self.expsk);
+            let mut h = Blake2bParams::new()
+                .hash_length(64)
```

Comment:
ZIP 32 specifies `BLAKE2b-256` here:
```suggestion
                .hash_length(32)
```
This bug should be caught by test vectors.